validation/contracts/ValidationEngine.sol

```diff
@@ -0,0 +1,45 @@
+import "./ValidationRuleInterface.sol";
+import "./ValidationStatus.sol";
+
+contract ValidationEngine is ValidationStatus {
+
+  mapping (bytes32 => bytes32[]) ruleNames;
+  mapping (bytes32 => mapping(bytes32 => mapping(bool => ValidationRuleInterface))) public ruleSets;
+
+  function addRule(bytes32 profileName, bytes32 ruleName, ValidationRuleInterface ruleContractAddress) public {
+    ruleSets[profileName][ruleName][true] = ruleContractAddress;
+
+    bytes32[] names = ruleNames[profileName];
+    names.push(ruleName);
+    ruleNames[profileName] = names;
+  }
+
+  function deactivateRule(bytes32 profileName, bytes32 ruleName) public {
+    ruleSets[profileName][ruleName][false] = ruleSets[profileName][ruleName][true];
+    ruleSets[profileName][ruleName][true]  = ValidationRuleInterface(0);
+  }
+
+  function activateRule(bytes32 profileName, bytes32 ruleName) public {
+    ruleSets[profileName][ruleName][true]  = ruleSets[profileName][ruleName][false];
+    ruleSets[profileName][ruleName][false] = ValidationRuleInterface(0);
+  }
+
+  function validate(address contractAddress, bytes32 profileName) public returns (bool, uint, bytes32) {
```

Comment:
please change to rest status:
`return (RestStatus.BAD_REQUEST, WingsError.PROFILE_NOT_FOUND, null);`
or
`return (RestStatus.OK, WingsError.NULL, payload);`